crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -266,9 +266,32 @@ impl Settlement {
         eth: &Ethereum,
         auction: &competition::Auction,
         revert_protection: &mempools::RevertProtection,
-    ) -> Result<competition::Score, boundary::Error> {
-        self.boundary
-            .score(eth, auction, self.gas.estimate, revert_protection)
+    ) -> Result<competition::Score, score::Error> {
+        let objective_value = self
+            .boundary
+            .objective_value(eth, auction, self.gas.estimate)?;
+
+        let score = match self.boundary.score() {
+            competition::SolverScore::Solver(score) => competition::Score(score),
+            competition::SolverScore::RiskAdjusted(success_probability) => competition::Score::new(
+                auction.score_cap(),
+                revert_protection,
+                objective_value,
+                self.gas.estimate,
+                auction.gas_price(),
+                success_probability,
+            )?,
+        };
+
+        if score > objective_value.into() {
+            return Err(score::Error::ScoreHigherThanObjective);
+        }
+
+        if score == eth::U256::zero().into() {
```

Comment:
nit: I think this is a thing
```suggestion
        if score.0.is_zero() {
```

Reply:
Tnx